### ISSUE TITLE
Hubmanager breaks b/c SDK doesn't support 5 levels of nested Twin

### ISSUE THREAD

**Issue by dluc** [Azure/iothub-manager-java#55]
# Type of issue

- [x] Bug
- [ ] New feature
- [ ] Enhancement

# Description

```
error] Test com.microsoft.azure.iotsolutions.iothubmanager.services.DevicesTest.getAsyncSuccessTest failed: java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: Malformed Json: exceed 5 levels, took 0.141 sec
[error]     at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)
[error]     at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1895)
[error]     at com.microsoft.azure.iotsolutions.iothubmanager.services.DevicesTest.getAsyncSuccessTest(DevicesTest.java:106)
[error]     ...
[error] Caused by: java.lang.IllegalArgumentException: Malformed Json: exceed 5 levels
[error]     at com.microsoft.azure.sdk.iot.deps.serializer.TwinParser.validateMap(TwinParser.java:1061)
[error]     at com.microsoft.azure.sdk.iot.deps.serializer.TwinParser.validateMap(TwinParser.java:1057)
[error]     at com.microsoft.azure.sdk.iot.deps.serializer.TwinParser.validateMap(TwinParser.java:1057)
[error]     at com.microsoft.azure.sdk.iot.deps.serializer.TwinParser.validateMap(TwinParser.java:1057)
[error]     at com.microsoft.azure.sdk.iot.deps.serializer.TwinParser.validateMap(TwinParser.java:1057)
[error]     at com.microsoft.azure.sdk.iot.deps.serializer.TwinParser.validateMap(TwinParser.java:1057)
[error]     at com.microsoft.azure.sdk.iot.deps.serializer.TwinParser.validateMap(TwinParser.java:1057)
[error]     at com.microsoft.azure.sdk.iot.deps.serializer.TwinParser.validateJson(TwinParser.java:1018)
[error]     at com.microsoft.azure.sdk.iot.deps.serializer.TwinParser.updateTwin(TwinParser.java:616)
[error]     at com.microsoft.azure.sdk.iot.service.devicetwin.DeviceTwin.getNextDeviceTwin(DeviceTwin.java:454)
[error]     at com.microsoft.azure.iotsolutions.iothubmanager.services.Devices.GetTwinByQueryAsync(Devices.java:251)
[error]     at com.microsoft.azure.iotsolutions.iothubmanager.services.Devices.lambda$queryAsync$1(Devices.java:88)
[error]     at java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:822)
[error]     at java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:797)
[error]     at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
[error]     at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1962)
[error]     at com.microsoft.azure.sdk.iot.service.RegistryManager.lambda$getDevicesAsync$2(RegistryManager.java:289)
[error]     at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error]     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[error]     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[error]     ... 1 more
```

**Comment by zhang-hua** [comment]
This is caused by the limitation of current Java implementation of TwinParser which defines [MAX_MAP_LEVEL=5](https://github.com/Azure/azure-iot-sdk-java/blob/master/deps/src/main/java/com/microsoft/azure/sdk/iot/deps/serializer/TwinParser.java#L27) and throw IllegalArgumentException if level is greater than 5. 
I have created an issue [158](https://github.com/Azure/azure-iot-sdk-java/issues/158) for JAVA SDK.

Here is an example of failing twin reported property(4 levels based on document description):
```
"reported": { 
        "Protocol": "MQTT",
        "SupportedMethods": "Reboot,FirmwareUpdate,EmergencyValveRelease,IncreasePressure",
        "Telemetry": {                            -> Level 1
          "chiller-sensors;v1": {              -> Level 2
            "Interval": "00:00:10",
            "MessageTemplate": "{\"temperature\":${temperature},\"temperature_unit\":\"${temperature_unit}\",\"humidity\":${humidity},\"humidity_unit\":\"${humidity_unit}\",\"pressure\":${pressure},\"pressure_unit\":\"${pressure_unit}\"}",
            "MessageSchema": {             -> Level 3
              "Name": "chiller-sensors;v1",
              "Format": "JSON",
              "Fields": {                             -> Level 4
                "temperature": "Double", 
                "temperature_unit": "Text",
                "humidity": "Double",
                "humidity_unit": "Text",
                "pressure": "Double",
                "pressure_unit": "Text"
              }
            }
          }
        }
```

**Comment by dluc** [comment]
@zhang-hua can you make the test resilient to this issue? we're going to share resources between Java and .NET very often, and we cannot afford to have broken builds when there is no bug in the code

**Comment by zhang-hua** [comment]
Fixed in PR #64 and close it.
